src/Microsoft.SqlTools.ServiceLayer/Rename/RenameUtils.cs

```diff
@@ -0,0 +1,74 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using System;
+using System.Text.RegularExpressions;
+using Microsoft.Data.SqlClient;
+using Microsoft.SqlServer.Management.Common;
+using Microsoft.SqlServer.Management.Sdk.Sfc;
+using Microsoft.SqlServer.Management.Smo;
+using Microsoft.SqlTools.ServiceLayer.Rename.Requests;
+
+namespace Microsoft.SqlTools.ServiceLayer.Rename
+{
+    /// <summary>
+    /// Helper class for Rename Service
+    /// </summary>
+    public static class RenameUtils
+    {
+        private static readonly string UrnPrefix = "Server[@Name='{0}']/Database[@Name='{1}']/Table[@Name='{2}' and @Schema='{3}']";
+        private static readonly string RegexMatchPatternValidObjectName = @"^[\p{L}_][\p{L}\p{N}@$#_]{0,127}$";
+        /// <summary>
+        /// Method to validate all the parameters for the renaming operation
+        /// </summary>
+        /// <param name="requestParams">parameters which should be checked</param>
+        public static void Validate(ProcessRenameEditRequestParams requestParams)
+        {
+            if (requestParams == null)
+            {
+                throw new ArgumentNullException();
+            }
+            if (String.IsNullOrEmpty(requestParams.TableInfo.TableName) || String.IsNullOrEmpty(requestParams.TableInfo.Database) || String.IsNullOrEmpty(requestParams.TableInfo.Schema) || String.IsNullOrEmpty(requestParams.TableInfo.OldName) || String.IsNullOrEmpty(requestParams.ChangeInfo.NewName))
+            {
+                throw new ArgumentException(SR.RenameRequestParametersNotNullOrEmpty);
+            }
+            if (!Regex.IsMatch(requestParams.TableInfo.Schema, RegexMatchPatternValidObjectName) || !Regex.IsMatch(requestParams.TableInfo.Database, RegexMatchPatternValidObjectName) || !Regex.IsMatch(requestParams.TableInfo.OldName, RegexMatchPatternValidObjectName) || !Regex.IsMatch(requestParams.TableInfo.TableName, RegexMatchPatternValidObjectName) || !Regex.IsMatch(requestParams.ChangeInfo.NewName, RegexMatchPatternValidObjectName))
+            {
+                throw new ArgumentOutOfRangeException(SR.NotAllowedCharacterForRenaming);
```

Comment:
We should just let SMO/the engine do all the logic for this I think, not try to re-implement any of that ourselves.

Reply:
Surely I can perfectly remove it 😊

Reply:
I'd remove the whole method